misc/python/materialize/cli/mzcompose.py

```diff
@@ -31,117 +29,99 @@
 MIN_COMPOSE_VERSION = (1, 24, 0)
 
 
-def assert_docker_compose_version() -> None:
-    """Check the version of docker-compose installed.
-
-    Raises `MzRuntimeError` if the version is not recent enough.
-    """
-    cmd = ["docker-compose", "version", "--short"]
-    output = spawn.capture(cmd, unicode=True).strip()
-    version = tuple(int(i) for i in output.split("."))
-    if version < MIN_COMPOSE_VERSION:
-        msg = f"Unsupported docker-compose version: {version}, min required: {MIN_COMPOSE_VERSION}"
-        raise errors.MzConfigurationError(msg)
-
-
 def main(argv: List[str]) -> int:
     # Lightly parse the arguments so we know what to do.
     args, unknown_args = ArgumentParser().parse_known_args(argv)
-    if not args.file:
-        config_files = ["mzcompose.yml"]
-    else:
-        config_files = args.file
+    if args.file:
+        raise errors.MzConfigurationError("-f/--file option not supported")
+    elif args.project_directory:
+        raise errors.MzConfigurationError("--project-directory option not supported")
 
     ui.Verbosity.init_from_env(args.mz_quiet)
 
+    # Load repository.
     root = Path(os.environ["MZ_ROOT"])
     repo = mzbuild.Repository(root)
 
+    # Handle special mzcompose commands that apply to the repo.
     if args.command == "gen-shortcuts":
         return gen_shortcuts(repo)
-
-    assert_docker_compose_version()
+    elif args.command == "list-compositions":
+        for name in repo.compositions:
+            print(name)
+        return 0
+
+    # Load composition.
+    try:
+        composition = mzcompose.Composition(repo, args.mz_find or Path.cwd().name)
+    except errors.UnknownComposition:
+        if args.mz_find:
+            print(f'unknown composition "{args.mz_find}"', file=sys.stderr)
```

Comment:
TIOLI: it's slightly more robust to force a `repr` format with `!r`:

```suggestion
            print(f'unknown composition {args.mz_find!r}', file=sys.stderr)
```

```python
>>> v = 'one"\'two'
>>> print(f"{v!r}")
'one"\'two'
>>> c = 'three four'
>>> print(f"{c!r}")
'three four'
```

Reply:
ah, done.